Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -2523,18 +2541,67 @@ namespace Babylon
                         auto inputSourceVal = inputSourceFound->second.Value();
                         SetXRInputSourceData(inputSourceVal, inputSource);
 
-                        //inputSource already exists, find the corresponding gamepad object if enabled and set to correct values
+                        // inputSource already exists, find the corresponding gamepad object if enabled and set to correct values
                         if (inputSourceVal.Has("gamepad"))
                         {
                             auto gamepadObject = inputSourceVal.Get("gamepad").As<Napi::Object>();
                             SetXRGamepadObjectData(inputSourceVal, gamepadObject, inputSource);
                         }
                     }
+
+                    // Handle gestures. Sources that do not support a gesture will always return false
+                    bool isSelected = inputSource.GamepadObject.Buttons[0].Pressed;
```

Comment:
is this gamepad object referring to motion controller?

Reply:
Ah, I was tempted to rename it. I'm using the Gamepad object to hold Select, Squeeze for all inputs, not just the motion controller it was originally used for.

Reply:
I didn't want to cloud XR.h with different types specific to OpenXR's options.

Reply:
nvm, i just realized this is the gamepadobject on the native end.